app/src/main/java/org/simple/clinic/newentry/PatientEntryScreenController.kt

```diff
@@ -126,32 +128,37 @@ class PatientEntryScreenController @Inject constructor(
         .map { showPattern -> { ui: Ui -> ui.setShowDatePatternInDateOfBirthLabel(showPattern) } }
   }
 
-  private fun ongoingPatientEntryUpdates(events: Observable<UiEvent>): Observable<OngoingPatientEntryChanged> {
-    val nameChanges = events
-        .ofType<PatientFullNameTextChanged>()
-        .map { it.fullName }
+  private fun mergeWithOngoingPatientEntryUpdates(): ObservableTransformer<UiEvent, UiEvent> {
+    return ObservableTransformer { events ->
 
-    val dateOfBirthChanges = events
-        .ofType<PatientDateOfBirthTextChanged>()
-        .map { it.dateOfBirth }
+      val personDetailChanges = personalDetailChanges(events)
 
-    val ageChanges = events
-        .ofType<PatientAgeTextChanged>()
-        .map { it.age }
+      val phoneNumberChanges = events
+          .ofType<PatientPhoneNumberTextChanged>()
+          .map { (phoneNumber) -> if (phoneNumber.isBlank()) None else Just(PhoneNumber(phoneNumber)) }
 
-    val genderChanges = events
-        .ofType<PatientGenderChanged>()
-        .map { it.gender }
+      val addressChanges = addressChanges(events)
 
-    val personDetailChanges = Observables.combineLatest(nameChanges, dateOfBirthChanges, ageChanges, genderChanges)
-    { name, dateOfBirth, age, gender ->
-      PersonalDetails(name, dateOfBirth.nullIfBlank(), age.nullIfBlank(), gender.toNullable())
-    }
+      val ongoingPatientEntryChanges = Observables
+          .combineLatest(
+              personDetailChanges,
+              phoneNumberChanges,
+              addressChanges
+          ) { personal, phone, address ->
+            OngoingPatientEntryChanged(OngoingNewPatientEntry(personal, address, phone.toNullable()))
+          }
+          .flatMapSingle { changedPatientEntry ->
+            patientRepository.ongoingEntry()
+                .map { alreadyPresentEntry ->
+                  changedPatientEntry.copy(entry = changedPatientEntry.entry.copy(identifier = alreadyPresentEntry.identifier))
```

Comment:
I was trying to simplify this code further.
Right now this block of code is a little confusing to read. 
I tried doing something like following:
```
 val identifiers = events
          .ofType<ScreenCreated>()
          .flatMapSingle { patientRepository.ongoingEntry() }
          .map { it.identifier.toOptional() }
```
and then combining with the rest of it:
```
...
 identifiers
          ) { personal, phone, address, identifier ->
            OngoingPatientEntryChanged(OngoingNewPatientEntry(personal, address, phone.toNullable(), identifier.toNullable()))
          }
```
But the other tests are failing and it would require changing the tests.
Can you take a look at this if possible?

Reply:
Let me know if this requires pairing.

Reply:
The tests are failing because you are not emitting the `ScreenCreated` event in the test. This change looks good right now, but I realised when working with this file that we need to move `OngoingNewPatientEntry` out of `PatientRepository` and persist it to disk as a `SharedPreference`.

This is because this is a model that needs to survive process restart. So this entire method and the `OngoingPatientEntryChanged` event are going to be removed in the near future. I'd like to leave this method as is for now.

Reply:
https://www.pivotaltracker.com/story/show/165442090